packages/vite/src/node/utils.ts

```diff
@@ -311,20 +317,10 @@ export function removeRawQuery(url: string): string {
   return url.replace(rawRE, '$1').replace(trailingSeparatorRE, '')
 }
 
-const replacePercentageRE = /%/g
 export function injectQuery(url: string, queryToInject: string): string {
-  // encode percents for consistent behavior with pathToFileURL
-  // see #2614 for details
-  const resolvedUrl = new URL(
-    url.replace(replacePercentageRE, '%25'),
```

Comment:
It works by simply removing this `url.replace`, but I think we can use `splitFileAndPostfix` and avoid `new URL` here.